### PR TITLE
[agg] Add metric tracking duration between writes

### PR DESCRIPTION
**Pull request by ryanhall07**
This replaces the previous attempt of recording the distributions of
expirations. The previous attempt would update the distribution on
ticks, which would heavily weight untouched metrics.

Instead we just want the distribution between writes to a metric, only
counting metrics that are actually updated.

**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

**Review by arnikola — APPROVED** [review]
LGTM!

File: src/aggregator/aggregator/entry.go [hunk 1]
```diff
@@ -1210,6 +1202,12 @@ func (e *Entry) applyValueRateLimit(numValues int64, m rateLimitEntryMetrics) er
 	return errWriteValueRateLimitExceeded
 }
 
+func (e *Entry) setLastAccessed(category metricCategory) {
+	now := e.nowFn().UnixNano()
+	e.metrics.durationBetweenWrites[category].RecordDuration(time.Duration(now - e.lastAccessNanos.Load()))
+	e.lastAccessNanos.Store(now)
```
Comment (prashantv) [inline src/aggregator/aggregator/entry.go:1208]: (optional nit) using `last := e.lastAccessNanos.Swap(now)` will batch the load+store operations

Comment (ryanhall07) [reply src/aggregator/aggregator/entry.go:1208]: oh nice! will do in a follow up

Comment (ryanhall07) [reply src/aggregator/aggregator/entry.go:1208]: https://github.com/m3db/m3/pull/3972

**Pull request merged**